### PR TITLE
Collab serial ble ot

### PR DESCRIPTION
**Pull request by canisLupus1313**
Merge of protocol serialization to main.

test-sdk-nrf: sdk-nrf-pr-16598

### CODE REVIEW

File: subsys/mpsl/cx/CMakeLists.txt
Comment (Damian-Nordic) [inline subsys/mpsl/cx/CMakeLists.txt:7]: 3WIRE got replaced with THREAD

File: subsys/mpsl/cx/Kconfig
Comment (Damian-Nordic) [inline subsys/mpsl/cx/Kconfig:66]: Merge issue - reverted Thread Radio Coexistence

File: subsys/mpsl/cx/Kconfig
Comment (Damian-Nordic) [inline subsys/mpsl/cx/Kconfig:75]: Merge issue - reverted SOC_SERIES_XXX

File: subsys/nrf_rpc/Kconfig
Comment (Damian-Nordic) [inline subsys/nrf_rpc/Kconfig:31]: nit:
```suggestion
	bool "UART transport for nRF RPC"
```
or "nRF RPC over UART" to be consistent with IPC wording.

File: CODEOWNERS
Comment (alxelax) [inline CODEOWNERS:1]: I guess it is worth to add ownership for ps_client sample here too: https://github.com/nrfconnect/sdk-nrf/pull/16598/files#diff-fcf14c4b7b34fe7a11916195871ae66a59be87a395f28db73e345ebdc828085bR182

File: .github/test-spec.yml
Comment (canisLupus1313) [reply .github/test-spec.yml:556]: fixed.

File: CODEOWNERS
Comment (canisLupus1313) [reply CODEOWNERS:1]: fixed

File: samples/nrf_rpc/entropy_nrf53/cpuapp/src/entropy_ser.c
Comment (canisLupus1313) [reply samples/nrf_rpc/entropy_nrf53/cpuapp/src/entropy_ser.c]: removed.

File: subsys/nrf_rpc/Kconfig
Comment (canisLupus1313) [reply subsys/nrf_rpc/Kconfig]: fixed.

File: subsys/mpsl/cx/Kconfig
Comment (canisLupus1313) [reply subsys/mpsl/cx/Kconfig:88]: fixed.

File: subsys/mpsl/cx/CMakeLists.txt
Comment (canisLupus1313) [reply subsys/mpsl/cx/CMakeLists.txt]: fixed.

**Review by Damian-Nordic — COMMENTED** [review]
I care mostly about the MPSL merge issues that are still not resolved. The remaining comments are optional and can be fixed after merge.

**Comment by tomaszkob89** [comment]
Below line will replace the default [test-fw-nrfconnect-nrf](https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-nrf/browse) repository branch used for tests on this pull request.

test-sdk-nrf: sdk-nrf-pr-16598

File: include/mpsl/mpsl_cx_software.h
Comment (alxelax) [inline include/mpsl/mpsl_cx_software.h:50]: No description of returned parameter.

**Review by alxelax — CHANGES_REQUESTED** [review]
All new functionality, improvements, additions, application, and sample should be added into release notes.

**Review by Damian-Nordic — APPROVED** [review]
I agree about release notes but other than that, looks good to me.

**Comment by canisLupus1313** [comment]
@alxelax I have applied your suggestions please check.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (alxelax) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:506]:
```suggestion
* Added Experimental serialization of Openthread APIs.
```

Comment (alxelax) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:506]: Also, please add the following:
```
Added the logging backend that sends logs as nRF RPC events.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:238]:
```suggestion
Protocols serialization
-----------------------

* Added the protocol serialization server application.
```
Let's move it to after the nRF Machine Learning (Edge Impulse) section to keep it in alphabetical order

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:359]:
```suggestion
* Added the protocol serialization client sample.
```
as above - let's align this section alphabetically

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:506]:
```suggestion
  * Added:
  
    * An experimental serialization of Openthread APIs.
    * The logging backend that sends logs through nRF RPC events.
```
this includes the above suggestions

**Review by annwoj — COMMENTED** [review]
Just few suggestions. Once the documentation work is complete it'd be good to add references to the current entries.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (Damian-Nordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:238]: nit: "Protocol" -> "Protocols" everywhere.

Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: fixed.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: fixed.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: fixed.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: fixed.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:347]:
```suggestion
* Added the protocols serialization client sample.
```
this wasn't fixed

**Review by annwoj — COMMENTED** [review]
just the last nit

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Please check again.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:224]:
```suggestion
* Added the Protocols serialization server application.
```
Sorry - removing that capitalization was unintended - since it's a name we should keep it capitalized

Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: no problem should be fine now.

**Comment by canisLupus1313** [comment]
@alxelax  Thanks.

**Review by annwoj — APPROVED** [review]
lgtm, thanks!

**Comment by canisLupus1313** [comment]
@annwoj Thanks.

File: samples/nrf_rpc/ps_client/src/ot_shell.c
Comment (SylwesterKonczyk) [inline samples/nrf_rpc/ps_client/src/ot_shell.c:1]: Wasn't it possible to reuse regular OpenThread CLI?

File: subsys/net/openthread/rpc/Kconfig
Comment (SylwesterKonczyk) [inline subsys/net/openthread/rpc/Kconfig:27]: Do we plan to keep that dependency in the future? Avoiding it would enable integrations in IP-stack-less environments, i.e. Customer would be able to rely on L7 API, i.e. COAP

File: samples/nrf_rpc/ps_client/src/ot_shell.c
Comment (Damian-Nordic) [reply samples/nrf_rpc/ps_client/src/ot_shell.c:1]: Then we would probably need to link OpenThread library on the client side, which I'm not sure is desired (?).
Also, using OpenThread CLI may imply that we have to have all core OpenThread functions already serializeable (or we get linker errors). But anyone sees a solution how we can selectively include OpenThread CLI implementation and think there's no issue with linking OpenThread libraries on the client, I would love it as well :)

File: subsys/net/openthread/rpc/Kconfig
Comment (Damian-Nordic) [reply subsys/net/openthread/rpc/Kconfig:27]: Only `if_client` file depends on it, so we can compile it conditionally based on `NETWORKING` instead.

Comment (canisLupus1313) [reply subsys/net/openthread/rpc/Kconfig:27]: This can be mitigated for future Since this is currently only option it has to stay. You can rais a Jira for reference.

File: samples/nrf_rpc/ps_client/src/ot_shell.c
Comment (canisLupus1313) [reply samples/nrf_rpc/ps_client/src/ot_shell.c:1]: Please raise  a jira ticket  for this kind of discussion.

Comment (alxelax) [reply samples/nrf_rpc/ps_client/src/ot_shell.c:1]: I guess this requires some discussion since it is not clear how OpenThread CLI is going to coexist with Zephyr's shell in case of OT and BLE on the same device.

**Review by SylwesterKonczyk — APPROVED** [review]

File: samples/nrf_rpc/ps_client/sample.yaml
Comment (katgiadla) [inline samples/nrf_rpc/ps_client/sample.yaml:1]: Is any loopback required to run this sample? If yes, please add information about the obligatory use of the fixture.

Comment (alxelax) [reply samples/nrf_rpc/ps_client/sample.yaml:1]: Hi @katgiadla, documentation regarding the sample is in separate PR: https://github.com/nrfconnect/sdk-nrf/pull/16597
it is blocked until this is merged. Please take a look there if you can find there anything regarding your question.

Comment (canisLupus1313) [reply samples/nrf_rpc/ps_client/sample.yaml:1]: it's not needed.

**Pull request merged**
